plagiarism/process.go

```diff
@@ -0,0 +1,136 @@
+package plagiarism
+
+import (
+    "io"
+    "io/ioutil"
+    "encoding/json"
+    "log"
+    "github.com/SpaceWhite/gomoss"
+    "fmt"
+    "os"
+)
+
+// GUIDsStruct use for decoding JSON
+type GUIDsStruct struct {
+    GUIDs []string
+}
+
+// Analysis struct contains output dir and reference dir
+type Analysis struct {
+    GIUDs []string
+    SubmissionDir string
+    Mossid uint32
+    Logger *log.Logger
+    ResultDir string
+}
+
+// DecodeRequest decode request from request body
+func DecodeRequest(body io.ReadCloser) ([]string, error) {
+    decoder := json.NewDecoder(body)
+    var guidsStruct GUIDsStruct
+    err := decoder.Decode(&guidsStruct)
+
+    if err != nil {
+        return []string{}, err
+    }
+    return guidsStruct.GUIDs, nil
+}
+
+// GUIDPath use to get submission from OmegaUp
+// Example GUID : abcdefghi1234
+// GUID Path    : submissionDir/ab/cdefghi1234
+func (analysis *Analysis) GUIDPath(guid string) string {
+    return fmt.Sprintf("%v/%v/%v", analysis.SubmissionDir, guid[:2], guid[2:])
+}
+
+// GUIDResultDir use to specify in what directory the analysis results are stored
+// Example GUID : abcdefghi1234
+// GUID Path    : resultDir/ab
+func (analysis *Analysis) GUIDResultDir(guid string) string {
+    return fmt.Sprintf("%v/%v", analysis.ResultDir, guid[:2])
+}
+
+// GUIDResultPath use to specify in what path the analysis results are stored
+// Example GUID : abcdefghi1234
+// GUID Path    : resultDir/abcdefghi1234
+func (analysis *Analysis) GUIDResultPath(guid string) string {
+    return fmt.Sprintf("%v/%v/%v", analysis.ResultDir, guid[:2], guid[2:])
+}
+
+// RunDetector requests MOSS server to run plagiarism detector
+func (analysis *Analysis) RunDetector() Response {
+    m := gomoss.New(analysis.Mossid)
+    response := Response{
```

Comment:
It's a bit confusing that I need to switch between this file and `http.go` to see what's being used. How about limiting yourself such that this file does not use anything from `http.go`?

You can think of it this way: the more the files in the code approach a [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph), the better. In other words, dependency cycles are bad.